### ISSUE TITLE
Mixxx 2.4.1 crashing upon opening in Linux Penguin - Flatpak installed

### ISSUE THREAD

**Issue by hh37138** [mixxxdj/mixxx#13720]
### Bug Description

Every time I attempt to open the program in Linux on my Chromebook, it force closes immediately. The program worked fine a couple of days ago. Today I attempted to install a new USB Mic (it didn't work even in Chrome) and I updated my Chrome system. No changes were made to Linux. I attempted to run the program from the terminal and uninstall and reinstall it and received the following: 
```
hollywoodhowiegaroutte@penguin:~$ flatpak run org.mixxx.Mixxx
F: Ignoring D-Conf migrate-path setting /org/mixxx/Mixxx
Using preferences ScaleFactor 1
Loading resources from  "/app/share/mixxx/"
Configuration file is at the current version "2.4.1"
BroadcastSettings - Found 1 profile(s)
 qt.qpa.wayland:warning [Main] Ignoring unexpected wl_surface.enter received for output with id: 7 screen name: "Screen5" screen model: "unknown" This is most likely a bug in the compositor.
warning [Main] failed to load portaudio for JACK rename
ALSA lib ../../../src/pcm/pcm.c:2666:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.front
ALSA lib ../../../src/pcm/pcm.c:2666:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.rear
ALSA lib ../../../src/pcm/pcm.c:2666:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.center_lfe
ALSA lib ../../../src/pcm/pcm.c:2666:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.side
ALSA lib ../../../src/pcm/pcm.c:2666:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.surround21
ALSA lib ../../../src/pcm/pcm.c:2666:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.surround21
ALSA lib ../../../src/pcm/pcm.c:2666:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.surround40
ALSA lib ../../../src/pcm/pcm.c:2666:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.surround41
ALSA lib ../../../src/pcm/pcm.c:2666:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.surround50
ALSA lib ../../../src/pcm/pcm.c:2666:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.surround51
ALSA lib ../../../src/pcm/pcm.c:2666:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.surround71
ALSA lib ../../../src/pcm/pcm.c:2666:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.iec958
ALSA lib ../../../src/pcm/pcm.c:2666:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.iec958
ALSA lib ../../../src/pcm/pcm.c:2666:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.iec958
ALSA lib ../../../src/pcm/pcm.c:2666:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.hdmi
ALSA lib ../../../src/pcm/pcm.c:2666:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.hdmi
ALSA lib ../../../src/pcm/pcm.c:2666:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.modem
ALSA lib ../../../src/pcm/pcm.c:2666:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.modem
ALSA lib ../../../src/pcm/pcm.c:2666:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.phoneline
ALSA lib ../../../src/pcm/pcm.c:2666:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.phoneline
ALSA lib ../../oss/pcm_oss.c:404:(_snd_pcm_oss_open) Cannot open device /dev/dsp
ALSA lib ../../oss/pcm_oss.c:404:(_snd_pcm_oss_open) Cannot open device /dev/dsp
ALSA lib ../../src/confmisc.c:160:(snd_config_get_card) Invalid field card
ALSA lib ../../usb_stream/pcm_usb_stream.c:481:(_snd_pcm_usb_stream_open) Invalid card 'card'
ALSA lib ../../src/confmisc.c:160:(snd_config_get_card) Invalid field card
ALSA lib ../../usb_stream/pcm_usb_stream.c:481:(_snd_pcm_usb_stream_open) Invalid card 'card'
ALSA lib ../../../src/pcm/pcm_direct.c:2049:(snd1_pcm_direct_parse_open_conf) The field ipc_gid must be a valid group (create group audio)
warning [Main] Output sound device clock reference not set! Using "sysdefault"
warning [0x7f3724002760] Engine thread not scheduled with the real-time policy SCHED_FIFO
hollywoodhowiegaroutte@penguin:~$ flatpak uninstall --delete-data org.mixxx.Mixxx


        ID                            Branch        Op
 1. [✗] org.mixxx.Mixxx               stable        r
 2. [ ] org.mixxx.Mixxx.Locale        stable        r

Error: Flatpak system operation Uninstall not allowed for user
error: Failed to uninstall org.mixxx.Mixxx: Flatpak system operation Uninstall not allowed for user
hollywoodhowiegaroutte@penguin:~$ flatpak install org.mixxx.Mixxx
Looking for matches…
Remotes found with refs similar to ‘org.mixxx.Mixxx’:

   1) ‘flathub’ (system)
   2) ‘flathub’ (user)

Which do you want to use (0 to abort)? [0-2]: 1
Skipping: org.mixxx.Mixxx/x86_64/stable is already installed
```
ASK: Please review and advise on a solution to be able to use Mixxx.

### Version

2.4.1

### OS

Linux Penguin

**Comment by hh37138** [comment]
![Screenshot 2024-10-04 8 32 07 PM](https://github.com/user-attachments/assets/cfc761fc-80be-4746-bbab-7a51fbeaea2c)
 A lot of the code from the terminal appears to have been deleted, so here is a screenshot of it

**Comment by hh37138** [comment]
BTW, I also increased the Linux disk from 10 to 30 G with no change in behavior.

**Comment by daschuer** [comment]
It looks like something is messed up with your audio hardware. However this should be no excuse for a crash. 

Can you provide the mixxx.log file, which contains more debug information. 

You may also try to provide a gdb backtrace. It should be: 
`flatpak run --devel --command=sh org.mixxx.Mixxx`
`gdb ./mixxx --developer`
Make it crash ... 
`bt`
(Not tested)

**Comment by Swiftb0y** [comment]
> warning [Main] failed to load portaudio for JACK rename

I think its rather crashing because it can't load portaudio

https://github.com/mixxxdj/mixxx/blob/60bdaca11e4011121895a59bbf2e05549020de63/src/soundio/soundmanager.cpp#L654-L677

**Comment by hh37138** [comment]
> It looks like something is messed up with your audio hardware. However this should be no excuse for a crash.
> 
> Can you provide the mixxx.log file, which contains more debug information.
> 
> You may also try to provide a gdb backtrace. It should be:
> 
> ```
> flatpak run --devel --command=sh org.mixxx.Mixxx
> gdb ./mixxx --developer
> ```
> 
> Make it crash ... `bt` (Not tested)

Daschuer, How do I locate the mixxx log file when I can't open mixxx? I'm brand new to Linux and have practically no knowledge of how to use it. 

Here is a screenshot of what happened when I tried to do the gdb backtrace

![Screenshot 2024-10-05 5 27 57 PM](https://github.com/user-attachments/assets/55e26feb-cae1-4099-b4f0-dd95ab192fd4)

**Comment by hh37138** [comment]
> > warning [Main] failed to load portaudio for JACK rename
> 
> I think its rather crashing because it can't load portaudio

Swiftb0y, 

The only way I know how to change from Jack to Alsa (?) is inside of Mixxx. Is there a way to do that from the terminal? My old microphone set-up would not work with Jack, but it would with the other option in Mixxx. I may have switched it over to Jack as part of my troubleshooting.

**Comment by ronso0** [comment]
> Here is a screenshot of what happened when I tried to do the gdb backtrace

This tells you gdb is not installed. Install it with `sudo apt-get install gdb`
Complete steps to create backtraces are here:
https://github.com/mixxxdj/mixxx/wiki/Creating-Backtraces#linux--mac-os-x-with-xcode-4x

> The only way I know how to change from Jack to Alsa (?) is inside of Mixxx

Take a look a your Mixxx settings directory (here's [how to find it](https://manual.mixxx.org/2.4/en/chapters/appendix/settings_directory.html)). There should be a `soundconfig.xml` which holds the sound configuration incl. the API.
Rename it and start Mixxx.

**Comment by hh37138** [comment]
@ronso0 Thank you! I'm learning more about Linux every day. 

I was able to install gdb using the Sudo command line you provided, however, when I tried to run it again it failed. 
![image](https://github.com/user-attachments/assets/7b502d5f-0843-4c78-93cf-ff7e889d69c3)
I went into my Linux Program Files and wasn't able to locate the soundconfig.xml file 
![image](https://github.com/user-attachments/assets/2760f32b-1f6a-4230-b092-d2b096515981).

@Swiftb0y @daschuer @ronso0 Thank you all for your help so far. I appreciate it!

**Comment by daschuer** [comment]
>  I think its rather crashing because it can't load portaudio

The attempt to set the Jack client name is not the case for the crash. It is only an optional convenience feature to show Mixxx as Mixxx and not as portaudio in Jack tools. 

It however indicates that portaudio is not loaded as "libportaudio.so.2". Does it use a different name or static linking?

**Comment by daschuer** [comment]
@hh37138 You have a typo in the `gdb --help` command. It is gdb with b for bug. 

Please start gdb without extra command line options (sorry fro the wrong hint). 
`gbd ./mixxx` 
And then inside gdb
`run --developer`

**Comment by daschuer** [comment]
It looks like jack is not installed with flatpack: 

https://github.com/flathub/org.mixxx.Mixxx/blob/07271bed214b6e1cd5eb98c6c0dab9304ac61d49/org.mixxx.Mixxx.yaml#L148

@FakeShemp @djantti 

Is jack/pipewire working with flatpack? 
Can you add libjack.so ? 

Can you also confirm the situation with libportdio.so 
Is it available for dynamic loading? At which part is it located?

**Comment by hh37138** [comment]
@daschuer Thanks for the update! I got a little bit farther. 
![image](https://github.com/user-attachments/assets/29d7d51d-03cf-4728-a1be-9abea8019336)

**Comment by ronso0** [comment]
`gdb ./mixxx` means gdb should look for an executable called _mixxx_ in the current directory.
on the left of the command proper you can see the current working directory, and ~ means you're currently in your user home directory.l (you won't find mixxx there, unless you did a very unusual installation yourself).
You want the executable from /usr/bin (or some other default lookup path), so you need to type only 

`gdb mixxx`

then, as written in the wiki

`set height 0`
`run`
etc.

**Comment by hh37138** [comment]
@ronso0 
```
Take a look a your Mixxx settings directory (here's [how to find it](https://manual.mixxx.org/2.4/en/chapters/appendix/settings_directory.html)). There should be a soundconfig.xml which holds the sound configuration incl. the API.
Rename it and start Mixxx, which should then use the defaults (ALSA?).
```
Somehow I missed the direction in the wiki to make sure I could see all my hidden files in the folders. Once I enabled that I could find the soundconfig.xml file, and rename it. I was able to open Mixxx without it crashing!! It was still set on Jack, but I quickly reset it to Alsa. After saving my settings, I closed the app and reopened it with no issues! Thanks so much!! Bossman no longer wants to fire me! (He can't, we're married, LOL) 

I still haven't figured out the gdb thing. Let me know if you still need it for some other aspect of this ticket, but my inital concern is resolved  😂

**Comment by daschuer** [comment]
Is jack still listed in the API combo box of your hardware preferences? 

Mixxx should never crash not even with a faulty configuration. A backtrace will help to acheve this. So please try to produce one with the old config.

**Comment by hh37138** [comment]
Sorry for the long response time. The day job was insisting that I show up for once :)

I put Jack back as the API combo and it's still not crashing <3

I can't get the debug log to load from within mixxx or the Linux terminal. When attempting from inside Mixxx I click the button on the Library tab and nothing happens. 
[Screen recording 2024-10-10 3.10.29 PM.webm](https://github.com/user-attachments/assets/bb1d7caa-2c3e-4051-a609-845815773721)
Here are the Terminal results: 
![image](https://github.com/user-attachments/assets/3f0f34b5-f42a-4dcb-ba51-6e9728819012)
![image](https://github.com/user-attachments/assets/369aa647-ed87-4ff2-9ce7-e49cab11f527)

**Comment by ronso0** [comment]
The leading `./` tells the interpreter to look for the following file or command in the current working directory. The `~` before the prompt indicates this is your home directory, but you want the mixxx binary from one of the usual binary lookup paths (usr/bin for example)

TL;DR
instead of `./mixxx` use just `mixxx` (plus the other arguments, of course)

> I put Jack back as the API combo and it's still not crashing

I _think_ @daschuer meant to you put back the previous (crashing) soundconfig.xml, only with that the debug log is useful.
Could that still be in your Trash?

**Comment by hh37138** [comment]
soundconfig.xml still exists in my files. 

I only renamed the file to soundconfigure.xml, which stopped the crashing. I tried to rename it what it was, but the system said the file name still existed. 

I tried the code you suggested and the files were not found: 
![image](https://github.com/user-attachments/assets/9b814264-0f4a-424c-8391-8be3134ba68b)

**Comment by ronso0** [comment]
(please post copied text instead of screenshots, that allows us to copy it for responses, thanks)

Oh sorry, I forgot this is about the Flatpak version :facepalm: 
That explains why mixxx is not found (and why there is no .mixxx/mixxx.log I think).

For debugging a **Flatpak** app the procedure is a bit different:
* install the Flatpak debug sdk **org.freedesktop.Sdk.Debug**
`flatpak install org.freedesktop.Sdk.Debug/x86_64/23.08`
* run Mixxx in the debug sdk
`flatpak run --command=sh --devel org.mixxx.Mixxx --developer`
* proceed with with step 1 from here https://github.com/mixxxdj/mixxx/wiki/Creating-Backtraces#for-all-of-the-above

Note: I don't have the sdk installed, so I can't test these steps.

**Comment by hh37138** [comment]
Hi @ronso0, thank you so much for the directions! 

Here are the results of the flatpak debug install attempt: 

hollywoodhowiegaroutte@penguin:~$ flatpak install org.freedesktop.Sdk.Debug/x86_64/23.08
Looking for matches…
Remotes found with refs similar to ‘org.freedesktop.Sdk.Debug/x86_64/23.08’:

   1) ‘flathub’ (system)
   2) ‘flathub’ (user)

Which do you want to use (0 to abort)? [0-2]: 1


        ID                               Branch        Op       Remote        Download
 1. [✗] org.freedesktop.Sdk.Debug        23.08         i        flathub       2.8 GB / 2.5 GB

Warning: Failed to get revokefs-fuse socket from system-helper: Flatpak system operation GetRevokefsFd not allowed for user
Error: Flatpak system operation Deploy not allowed for user
^[[40;55Rerror: Failed to install org.freedesktop.Sdk.Debug: Flatpak system operation Deploy not allowed for user

**Comment by djantti** [comment]
> Is jack/pipewire working with flatpack? Can you add libjack.so ?
> 
> Can you also confirm the situation with libportdio.so Is it available for dynamic loading? At which part is it located?

Sorry for the delayed reply. As far as I know JACK is not not supported at all (and never will be) on Flatpaks and the only way of using it is through PipeWire. It does work really well nowadays. PortAudio is also included and loaded correctly.

But something super strange is happening in this bug report, since even the common Flatpak uninstall / install commands appear to fail. I'll see if I can get a ChromeOS VM going and dig around a bit.

**Comment by djantti** [comment]
Yeah, AFAIK that is the most common way of running somewhat standard Linux on Chromebooks.

ChromeOS (a highly locked down custom Linux distribution by itself) runs a virtual machine called crosvm and creates a container inside it with a mutable Linux environment. And then we have Mixxx running in a Flatpak sandbox on top of that.

I did manage to get ChromeOS running in a virtual machine, but it just states that 'Linux is not supported on your Chromebook'. Oh well... it prolly needs a bare metal install and / or a Google account.
